src/linker/Linker/Driver.cs

```diff
@@ -266,6 +267,12 @@ protected int SetupContext (ILogger customLogger = null)
 
 						continue;
 
+					case "--per-assembly-step":
```

Comment:
Does it need to be a new option?

Reply:
Not necessarily - but it makes it clear that this isn't the same as a normal `IStep`-based custom step. Some other options are:
- Add special syntax for `--custom-step` for the new kind of step
- Use the same exact syntax as `--custom-step` - then detect which interface the step implements, and look for the before/after step in the appropriate pipeline for that step kind

It sounds like you prefer the latter?